### PR TITLE
Extract a TaxonPresenter out of the taxon route

### PR DESCRIPTION
**Pull request by mobaig**
Makes the route implementation simpler. It focuses on request-handling
logic while the presenter assembles an object containing everything
needed to render the views.

### CODE REVIEW

**Comment by mobaig** [comment]
Note to reviewers: given it's a fairly large change, worth pulling the branch down and having a click around to see if anything's broken.

File: app/models/taxon_presenter.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    this.breadcrumb =  breadcrumbMaker.getBreadcrumbForTaxon([this.basePath]);
+  }
+
+  buildParent () {
+    this.parent = this.breadcrumb[this.breadcrumb.length - 1];
+  }
+
+  buildChildren () {
+    this.children = this.taxon.atozChildren();
+    for (var i = 0; i < this.children.length; i += 1) {
```
Comment (carvil) [inline app/models/taxon_presenter.js]: An alternative to this iteration, you could use the [`for...of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) notation here, e.g.:

```
for(const child of this.children) {
   child. ....
}
```

File: app/models/taxon_presenter.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  buildContent () {
+    this.content = {}
+    this.content.guidance = this.taxon.filterByHeading('guidance');
+  }
+
+  checkIfPenultimate () {
+    var findGrandchild = _.find(this.children, function (childTaxon) {
+      return childTaxon.children.length > 0;
+    });
```
Comment (carvil) [inline app/models/taxon_presenter.js]: We could perhaps remove the `lodash` dependency and use the array methods here together with the new function syntax as well. What do you think of something like:

```
checkIfPenultimate () {
  var ifPenultimate = (childTaxon) => childTaxon.children.length > 0;
  var findGrandchild = this.children.find(ifPenultimate);
```

File: app/models/taxon_presenter.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+"use strict";
+
+var _ = require('lodash');
```
Comment (carvil) [inline app/models/taxon_presenter.js]: I think we could delete this dependency, more details below.

File: app/models/taxon_presenter.js (hunk 1 above)
Comment (mobaig) [inline app/models/taxon_presenter.js]: Replaced with a `map`.

File: app/models/taxon_presenter.js (hunk 2 above)
Comment (mobaig) [inline app/models/taxon_presenter.js]: Replaced.

File: app/models/taxon_presenter.js (hunk 3 above)
Comment (mobaig) [inline app/models/taxon_presenter.js]: Gone.

**Pull request merged**